benches/bench.rs

```diff
@@ -1,331 +1,101 @@
-}
-
-impl RandomKeys {
-    fn new() -> Self {
-        RandomKeys { state: 0 }
-    }
-}
-
-impl Iterator for RandomKeys {
-    type Item = usize;
-    fn next(&mut self) -> Option<usize> {
-        // Add 1 then multiply by some 32 bit prime.
-        self.state = self.state.wrapping_add(1).wrapping_mul(3787392781);
-        Some(self.state)
-    }
-}
-
-// Just an arbitrary side effect to make the maps not shortcircuit to the non-dropping path
-// when dropping maps/entries (most real world usages likely have drop in the key or value)
-lazy_static::lazy_static! {
-    static ref SIDE_EFFECT: AtomicUsize = AtomicUsize::new(0);
-}
-
-#[derive(Clone)]
-struct DropType(usize);
-impl Drop for DropType {
-    fn drop(&mut self) {
-        SIDE_EFFECT.fetch_add(self.0, atomic::Ordering::SeqCst);
-    }
-}
-
-macro_rules! bench_suite {
-    ($bench_macro:ident, $bench_ahash_serial:ident, $bench_std_serial:ident,
-     $bench_ahash_highbits:ident, $bench_std_highbits:ident,
-     $bench_ahash_random:ident, $bench_std_random:ident) => {
-        $bench_macro!($bench_ahash_serial, AHashMap, 0..);
-        $bench_macro!($bench_std_serial, StdHashMap, 0..);
-        $bench_macro!(
-            $bench_ahash_highbits,
-            AHashMap,
-            (0..).map(usize::swap_bytes)
-        );
-        $bench_macro!(
-            $bench_std_highbits,
-            StdHashMap,
-            (0..).map(usize::swap_bytes)
-        );
-        $bench_macro!($bench_ahash_random, AHashMap, RandomKeys::new());
-        $bench_macro!($bench_std_random, StdHashMap, RandomKeys::new());
-    };
-}
-
-macro_rules! bench_insert {
-    ($name:ident, $maptype:ident, $keydist:expr) => {
-        #[bench]
-        fn $name(b: &mut Bencher) {
-            let mut m = $maptype::with_capacity_and_hasher(SIZE, Default::default());
-            b.iter(|| {
-                m.clear();
-                for i in ($keydist).take(SIZE) {
-                    m.insert(i, (DropType(i), [i; 20]));
-                }
-                black_box(&mut m);
-            });
-            eprintln!("{}", SIDE_EFFECT.load(atomic::Ordering::SeqCst));
-        }
-    };
-}
-
-bench_suite!(
-    bench_insert,
-    insert_ahash_serial,
-    insert_std_serial,
-    insert_ahash_highbits,
-    insert_std_highbits,
-    insert_ahash_random,
-    insert_std_random
-);
-
-macro_rules! bench_grow_insert {
-    ($name:ident, $maptype:ident, $keydist:expr) => {
-        #[bench]
-        fn $name(b: &mut Bencher) {
-            b.iter(|| {
-                let mut m = $maptype::default();
-                for i in ($keydist).take(SIZE) {
-                    m.insert(i, DropType(i));
-                }
-                black_box(&mut m);
-            })
-        }
-    };
-}
-
-bench_suite!(
-    bench_grow_insert,
-    grow_insert_ahash_serial,
-    grow_insert_std_serial,
-    grow_insert_ahash_highbits,
-    grow_insert_std_highbits,
-    grow_insert_ahash_random,
-    grow_insert_std_random
-);
```

Comment:
Please don't remove all our benchmarks, we need those to accurately measure the performance of hashbrown in different situations.

Reply:
@Amanieu Where do these come from? As when I test locally, all of these functions are not found.

Reply:
Just run `cargo bench` with a nightly toolchain.